cynic/src/maybe_undefined.rs

```diff
@@ -0,0 +1,87 @@
+use serde::{Deserialize, Serialize};
+
+use std::ops::{Deref, DerefMut};
+
+/// A wrapper around async-graphql's [`MaybeUndefined`](https://docs.rs/async-graphql/latest/async_graphql/types/enum.MaybeUndefined.html).
+///
+/// You can initialize it from:
+/// - `From<Option<T>>` will become T or null.
+/// - `Default` will become undefined.
+#[derive(Serialize, Deserialize, Copy, Clone, PartialEq, PartialOrd, Eq, Ord, Debug, Hash)]
+pub struct MaybeUndefined<T>(async_graphql::MaybeUndefined<T>);
```

Comment:
As I already mentioned can you implement MaybeUndefined here rather than pulling in async_graphql